File: Java_ChildrenCompetitionSockets/CompetitionServices/src/main/java/competition/services/ICompetitionServices.java
package competition.services;

import competition.model.Participant;
import competition.model.TestDTO;
import competition.model.User;

public interface ICompetitionServices {
    void login(User user, ICompetitionObserver client) throws CompetitionException;
    void logout(User user, ICompetitionObserver client) throws CompetitionException;
    User findUserByUsername(String username) throws CompetitionException;
    Participant findParticipantByUsername(String username) throws CompetitionException;
    void saveParticipant(String username, String name, int age, int testId) throws CompetitionException;
    void saveRelation(int idTest, int idParticipant) throws CompetitionException;
    TestDTO[] findAllTestDTOs() throws CompetitionException;
    Participant[] findAllParticipantsForTest(Integer id) throws CompetitionException;
}

File: CSharp_ChildrenCompetitionSockets/CSharp_ChildrenCompetitionSockets/services/ICompetitionServices.cs
using System;
using CSharp_ChildrenCompetitionGUI.model;

namespace services
{
    public interface ICompetitionServices
    {
        void login(User user, ICompetitionObserver client);
        void logout(User user, ICompetitionObserver client);
        User findUserByUsername(String username);
        Participant findParticipantByUsername(String username);
        void saveParticipant(String username, String name, int age, int testId);
        void saveRelation(int idTest, int idParticipant);
        User[] getLoggedUsers(User user);
        TestDTO[] findAllTestDTOs();
        Participant[] findAllParticipantsForTest(int id);
    }
}